src/search/engines/bingimages/bingimages.go

```diff
@@ -51,23 +51,25 @@ func Search(ctx context.Context, query string, relay *bucket.Relay, options engi
 		if err := json.Unmarshal([]byte(metadataS), &jsonMetadata); err != nil {
 			log.Error().
 				Err(err).
-				Msg("bingimages.Search() -> onHTML: failed to unmarshal metadata")
+				Str("jsonMetadata", metadataS).
+				Msgf("bingimages.Search() -> onHTML: failed to unmarshal metadata")
 			return
 		}
 
-		if jsonMetadata.Purl == "" || jsonMetadata.Murl == "" || jsonMetadata.Turl == "" {
+		if jsonMetadata.ImageURL == "" || jsonMetadata.PageURL == "" || jsonMetadata.ThumbnailURL == "" {
 			log.Error().
 				Str("engine", Info.Name.String()).
 				Str("jsonMetadata", metadataS).
-				Str("url", jsonMetadata.Purl).
-				Str("original", jsonMetadata.Murl).
-				Str("thumbnail", jsonMetadata.Turl).
-				Msg("bingimages.Search() -> onHTML: Couldn't find image URL")
+				Str("url", jsonMetadata.PageURL).
+				Str("original", jsonMetadata.ImageURL).
+				Str("thumbnail", jsonMetadata.ThumbnailURL).
+				Msg("bingimages.Search() -> onHTML: Couldn't find image, thumbnail, or page URL")
 			return
 		}
 
 		titleText := strings.TrimSpace(dom.Find(dompaths.Title).Text())
 		if titleText == "" {
+			// could also use the json data ("t" field)
```

Comment:
If it's always available let's switch to that so colly doesn't have to parse unnecessary HTML